cmd/zoekt-webserver/main.go

```diff
@@ -504,3 +510,70 @@ var (
 		Help: "The total number of errors from zoekt watchdog.",
 	})
 )
+
+func mustRegisterMemoryMapMetrics(l sglog.Logger) {
+	if runtime.GOOS != "linux" {
+		l.Debug(
+			"skipping registration: must be running on a linux-based operating system",
+			sglog.String("current_operating_system", runtime.GOOS),
+			sglog.String("desired_operating_system", "linux"))
+
+		return
+	}
+
+	fs, err := procfs.NewDefaultFS()
+	if err != nil {
+		panic(fmt.Sprintf("registering memory map metrics: failed to initialize proc FS: %s", err))
+	}
+
+	prometheus.MustRegister(prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+		Name: "zoekt_webserver_max_memory_map_count",
+		Help: "Upper limit on amount of memory mapped regions a process may have.",
+	}, func() float64 {
+		vm, err := fs.VM()
+		if err != nil {
+			l.Debug(
+				"failed to read VM statistics",
+				sglog.String("path", path.Join(procfs.DefaultMountPoint, "sys", "vm")),
+				sglog.String("error", err.Error()),
+			)
+
+			return 0
+		}
+
+		if vm.MaxMapCount == nil {
+			return 0
+		}
+
+		return float64(*vm.MaxMapCount)
+	}))
+
+	prometheus.MustRegister(prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+		Name: "zoekt_webserver_current_memory_map_count",
+		Help: "Amount of memory mapped regions this process is currently using.",
+	}, func() float64 {
+		self, err := fs.Self()
```

Comment:
this looks like something we only need to call once.

Reply:
I think you are correct. The procFS api didn't make it clear to me whether or not the underlying data would be stale if we reused the same filesystem struct between multiple call sites. However, I've now looked at the implementation and it seems like we don't have anything to worry about there. 

I'll refactor.